src/projects/EnsureThat/EnsureTypeExtensions.cs

```diff
@@ -2,27 +2,36 @@
 using System.Diagnostics;
 using System.Reflection;
 using EnsureThat.Extensions;
+using JetBrains.Annotations;
 
 namespace EnsureThat
 {
     public static class EnsureTypeExtensions
     {
         private static class Types
         {
+            [NotNull]
```

Comment:
Why here? Think that it should only be on public facing APIs.

Reply:
Indeed, if you don't want the assistance when developing w/in the library (which can be helpful for the public methods because then R# can suggest it on public methods that call this) then it would not be needed. I personally love R#'s nullability checks (wish the language had them natively but a plugin works sufficiently) to tell me when I've done something silly or an assumption I've made about nullability of something isn't actually enforced/correct.

So, there are benefits for developers within Ensure.That, but as it provides no use to the end customer this is 💯 up to you for what your style would want in your library 😃

Reply:
[oh, of note the reason it's on some weird things like Types is b/c I like to use the R# 'pessimistic' analysis mode when working in library code to make SURE I've validated my assumptions - e.g. if I say my method doesn't return null, but I call out to some BCL/other library method that might actually - it alerts me to check the contract of that method (either in code or in documents)]